### ISSUE TITLE
Jupyterhub SSL CipherSuites & Protocol using configurable-http-proxy

### ISSUE THREAD

**Issue by atnjqt** [jupyterhub/configurable-http-proxy#279]
I am running jupyterhub and serving publicly over SSL, but when I check on SSLLabs for this site but we are getting a B rating and not A+... 

This is because **cipher suite & protocols** are not enforced, is there a way to configure this from inside the jupyterhub_config.py file?

I don't see any information regarding cipher or protocol on the [Enabling SSL Encryption](https://jupyterhub.readthedocs.io/en/stable/getting-started/security-basics.html#enabling-ssl-encryption) page...
Really I want to just use configurable-http-proxy and _not_ NGINX, as also described in a discussion post [here](https://discourse.jupyter.org/t/do-i-need-nginx-if-im-okay-with-configurable-http-proxy/5667)...

I see [here](https://github.com/jupyterhub/configurable-http-proxy#command-line-options) that there are configurable-http-proxy flags (specifically, `--ssl-protocol` and `--ssl-ciphers`) as command line options.

When I run ```systemctl status jupyterhub.service``` I see the command is launched when starting jhub `421979 node /usr/local/bin/configurable-http-proxy --ip 10.x.x.x. --port 443 --api-ip 127.0.0.1 --api-port 8001 --error-target http://127.0.0.1:8081/hub/error --ssl-key /path/to/ssl.key --ssl-cert /path/to/ssl.crt` so it seems like I should be able to pass other flags.

**Comment by atnjqt** [comment]
So is it possible to get an A+ rating on SSLLabs hosting JupyterHub with configurable-http-proxy or does this require nginx or apache (as described here https://jupyterhub.readthedocs.io/en/stable/reference/config-proxy.html)

There certainly are cmdline options for hardening SSL configurations, it's just unclear how to pass these additional commands from JupyterHub config to proxy!

**Comment by minrk** [comment]
@atnjqt what version of node.js? Typically, node's defaults will get you at least an A as long as your node is recent. This is why it's fairly rare to set these and not often discussed in docs.

But you can set the CHP flags for these with JupyterHub. If your Hub is starting CHP, you would use:

```python
c.ConfigurableHTTPProxy.command = ['configurable-http-proxy', '--ssl-version=TLSv1_2', '--ssl-ciphers=ECDHE-RSA-AES128-GCM-SHA256,ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-RSA-AES256-GCM-SHA384,ECDHE-ECDSA-AES256-GCM-SHA384,DHE-RSA-AES128-GCM-SHA256,ECDHE-RSA-AES128-SHA256,DHE-RSA-AES128-SHA256,ECDHE-RSA-AES256-SHA384,DHE-RSA-AES256-SHA384,ECDHE-RSA-AES256-SHA256,DHE-RSA-AES256-SHA256,HIGH,!aNULL,!eNULL,!EXPORT,!DES,!RC4,!MD5,!PSK,!SRP,!CAMELLIA']
```

to match the defaults in more recent node [here](https://github.com/nodejs/node/commit/5755fc099f883293530406c423bda47414834057)

**Comment by atnjqt** [comment]
@minrk thank you for the note! In hindsight I am feeling a little embarrassed that I did not check on Node.JS version first given CHP is a part of node ...

As you anticipated, I was running NodeJS v10 which is far from the latest stable. I just upgraded to NodeJS v14 and this solved my protocol / cipher issues capping me at SSLLabs B rating.

To further hardened and get an A rating I followed instructions [here](https://jupyterhub.readthedocs.io/en/stable/troubleshooting.html#use-a-chained-ssl-certificate) for Jhub chained certificates. I now get an A rating, I will continue playing around to get an A+.

Thanks again everyone for helping me through this, excited to avoid the reverse proxy implementation now that CHP is hardened!
